Snowflake.Data.Tests/SFBaseTest.cs

```diff
@@ -116,33 +138,69 @@ public void Setup()
 
             if (testConfigs.TryGetValue("testconnection", out var testConnectionConfig))
             {
-                TestConfig = testConnectionConfig;
-                TestConfig.schema = TestConfig.schema + "_" + Guid.NewGuid().ToString().Replace("-", "_");
+                s_testConfig = testConnectionConfig;
+                s_testConfig.schema = s_testConfig.schema + "_" + Guid.NewGuid().ToString().Replace("-", "_");
             }
             else
             {
                 Assert.Fail("Failed to load test configuration");
             }
             
-            ModifySchema(TestConfig.schema, SchemaAction.CREATE);
+            ModifySchema(s_testConfig.schema, SchemaAction.CREATE);
         }
-
+        
         [OneTimeTearDown]
         public void Cleanup()
         {
-            ModifySchema(TestConfig.schema, SchemaAction.DROP);
+            ModifySchema(s_testConfig.schema, SchemaAction.DROP);
+        }
+        
+        [OneTimeSetUp]
+        private void SetupTestPerformance()
+        {
+            s_testPerformance = new Dictionary<string, TimeSpan>();
         }
 
-        private static string connectionString => string.Format(ConnectionStringFmt,
-            TestConfig.protocol,
-            TestConfig.host,
-            TestConfig.port,
-            TestConfig.account,
-            TestConfig.role,
-            TestConfig.database,
-            TestConfig.warehouse,
-            TestConfig.user,
-            TestConfig.password);
+        [OneTimeTearDown]
+        private void CreateTestTimeArtifact()
+        {
+            var os = "";
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                os = "windows";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                os = "linux";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                os = "macos";
+            }
+            
+            var resultText = "test;time\n";
+            foreach (var test in s_testPerformance)

```

Comment:
Can we rewrite it to something like `s_testPerformance.map(test => $"{test.Key};{time...}").join("\n")`?

Reply:
Sure, rewrote it.